src/transformers/modeling_blenderbot.py

```diff
@@ -0,0 +1,201 @@
+        decoder_input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, target_sequence_length)`, `optional`, defaults to :obj:`None`):
+        decoder_attention_mask (:obj:`torch.BoolTensor` of shape :obj:`(batch_size, tgt_seq_len)`, `optional`, defaults to :obj:`None`):
+        labels: (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`, `optional`, defaults to :obj:`None`):
+"""
+
+class BlenderbotConditionalGeneration(PretrainedBlenderbotModel):
+    def __init__(self, config:BlenderbotConfig):
+        super().__init__(config)
+        #self.config = config
+        self.shared = nn.Embedding(config.vocab_size, config.d_model, config.pad_token_id)        
+        self.encoder = BlenderEncoder(config, self.shared)
+        self.decoder = BlenderbotDecoder(config, self.shared)
+        self.init_weights()
+        
+    @add_start_docstrings_to_callable(BLENDERBOT_INPUTS_DOCSTRING)
+    def forward(self, 
+                input_ids, 
+                encoder_outputs=None, 
+                decoder_input_ids=None, 
+                attention_mask=None,
+                decodeer_attention_mask=None,
+                labels=None,
+                decoder_cached_states=None,
+                use_cache=False
+                ):
+        if encoder_outputs is None:
+            encoder_outputs = self.encoder(input_ids=input_ids,attention_mask=attention_mask)
+        assert isinstance(encoder_outputs, tuple)
+        if use_cache:
+            decoder_padding_mask, casual_mask = None, None
+        else:
+            decoder_input_ids, decoder_padding_mask, casual_mask = _prepare_bart_decoder_inputs(self.config,
+                                                                                                input_ids,
+                                                                                                decoder_input_ids=decoder_input_ids,
+                                                                                                causal_mask_dtype=self.shared.weight.dtype,
+                                                                                                decoder_padding_mask=decodeer_attention_mask
+                                                                                            )
+        assert decoder_input_ids is not None
+        decoder_outputs = self.decoder(decoder_input_ids, encoder_outputs[0],attention_mask,
+                                       decoder_padding_mask, decoder_causal_mask=casual_mask,
+                                       decoder_cashed_states=decoder_cached_states,
+                                       use_cache=use_cache)
+        decoder_outputs: Tuple = _filter_out_falsey_values(decoder_outputs)
+        assert isinstance(decoder_outputs[0], torch.Tensor)
+        encoder_outputs: Tuple = _filter_out_falsey_values(encoder_outputs)
+        outputs = decoder_outputs + encoder_outputs
+        scores = F.linear(outputs[0], self.shared.weight)
```

Comment:
I would prefer for a linear layer to save it as a ``self.lm_head = nn.Linear(config.d_model, config.vocab_size, bias=False)``. The weights will automatically be tied to self.shared thanks to the `tie_weights()` function in self.init_weights()` if you provide the correct `self.input_embeddings()` function as well. Then you also don't have to instantiate a `nn.Linear()` layer each time `self.output_embeddings()` is called. 

What do you think @sshleifer ?